blockly-core/core/utils/xml.js

```diff
@@ -17,6 +17,12 @@
  * limitations under the License.
  */
 
+/* global Blockly, goog */
+
+// Ignore "Expected a conditional expression and instead saw an
+// assignment" warnings.
+/* jshint -W084 */
```

Comment:
So, for this warning in particular, I get:

```bash
Running "jshint:files" task

Running "jshint:some" (jshint) task

   ../blockly-core/core/utils/xml.js
     46 |  for (var i = 0, block; block = blocks[i]; i++) {
                                                   ^ Expected a conditional expression and instead saw an assignment.
     84 |  for (x = 0; input = block.inputList[x]; x++) {
                                                 ^ Expected a conditional expression and instead saw an assignment.
     85 |    for (y = 0; title = input.titleRow[y]; y++) {
                                                  ^ Expected a conditional expression and instead saw an assignment.
    101 |  for (i = 0; input = block.inputList[i]; i++) {
                                                 ^ Expected a conditional expression and instead saw an assignment.
    260 |  for (var i = 0; xmlChild = xml.childNodes[i]; i++) {
                                                       ^ Expected a conditional expression and instead saw an assignment.
    339 |  for (var x = 0, xmlChild; xmlChild = xmlBlock.childNodes[x]; x++) {
                                                                      ^ Expected a conditional expression and instead saw an assignment.
    348 |    for (var y = 0, grandchildNode; grandchildNode = xmlChild.childNodes[y];
                                                                                    ^ Expected a conditional expression and instead saw an assignment.
    445 |  for (var x = 0, child; child = xmlBlock.childNodes[x]; x++) {
                                                                ^ Expected a conditional expression and instead saw an assignment.

>> 8 errors in 1 file
Warning: Task "jshint:some" failed. Use --force to continue.

Aborted due to warnings.
```

Assigning members in the iteration condition is certainly not unheard-of, and it's a technique used frequently throughout the blockly-core code; I see 60 instances of this error when running jshint against all of `/blockly-core/core`.